portal-web/test/functional/com/liferay/portalweb/paths/pathlib/uielements/Modal.path

```diff
@@ -17,7 +17,7 @@
 </tr>
 <tr>
 	<td>BODY_MESSAGE</td>
-	<td>xpath=(//*[contains(@class,'modal-dialog')]//*[contains(@class,'modal-body')]/p)[${index}]</td>
+	<td>//*[contains(@class,'modal-dialog')]//*[contains(@class,'modal-body')]/p[${index}] | //*[contains(@class,'modal-dialog')]//*[contains(@class,'modal-body')]/div</td>
```

Comment:
Hi @martonakarczub,

This way is wrong.

Would you mind modifying to `xpath=(//*[contains(@class,'modal-dialog')]//*[contains(@class,'modal-body')]/*[self::p or self::div])[${index}]`

cc @gaborlovasdotcom

Here is a doc introducing using index in xpath https://liferay.atlassian.net/wiki/spaces/ENGECHO/pages/1947205663/Using+index+in+XPath

Reply:
Hi @Tim-Cao 

*[contains(@class, 'modal-body')] is going to find two div tag but both are ancestors of all of the searched p tags (so it doesn't matter which is found as from there the p tags can be identified properly). There will be only one list returned from which p[index] can properly pick the right tag for comparison. 
The case described in the article only should be applied if in the middle of the path there is a branching.
So like
<div>
	<div>
		<p>...
	</div>
	<div>
		<p>...
	</div>
</div>
but here it is 
<div>
	<div>
		<p>...
	</div>
</div>
In info modal the div tag holds the text without further tags, so I find misleading to use *[self::p or self::div])[${index}] because it looks like that there are several div tags for texts. 
I pushed a new one what expresses a div tag without children.
Please share your thoughts about this one. Thank you!

Reply:
Hi @martonakarczub,
It works on my local. `xpath=(//*[contains(@class,'modal-dialog')]//*[contains(@class,'modal-body')]/*[self::p or self::div])[${index}]`
For danger, warning and info modals, 
![image](https://github.com/liferay-staging/liferay-portal/assets/52661397/8f9358ba-e4e6-493b-9064-27122d31a162)
![image](https://github.com/liferay-staging/liferay-portal/assets/52661397/b56a9f5a-bc2d-4f36-85de-1e7774f3859c)
![image](https://github.com/liferay-staging/liferay-portal/assets/52661397/f9594b0e-729f-4fef-b501-7a8acff125d8)